zerver/views/registration.py

```diff
@@ -51,6 +51,13 @@
 def check_prereg_key_and_redirect(request: HttpRequest, confirmation_key: str) -> HttpResponse:
     # If the key isn't valid, show the error message on the original URL
     confirmation = Confirmation.objects.filter(confirmation_key=confirmation_key).first()
+    # if the prereg user has status revoked the confirmation link should throw an error page
+    confirmation_status_obj = Confirmation.objects.get(confirmation_key=confirmation_key,
+                                                       type=confirmation.type)
+    prereg_user = confirmation_status_obj.content_object
+    revoked_value = confirmation_settings.STATUS_REVOKED
+    if prereg_user.status == revoked_value:
```

Comment:
In general, I think it's better just use `confirmation_settings.STATUS_REVOKED` directly, rather than assigning it to `revoked_value` first?

Reply:
I mean this is in all the places where we do this kind of thing, not just here, we should change them to using `realm_reactivation_link_error.html` directly.